pkg/segment/results/mresults/metricresults.go

```diff
@@ -659,6 +659,23 @@ func (r *MetricsResult) FetchPromqlMetricsForUi(mQuery *structs.MetricsQuery, pq
 			allTimestamps[ts] = struct{}{}
 		}
 	}
+	//add timestamps where metrics value is null
+	if calculatedInterval >= 600 {
+		t := time.Unix(int64(startTime), 0).UTC()
+		truncated := t.Truncate(time.Hour)
+		if t.Before(truncated) {
+			truncated = truncated.Add(time.Hour)
+		}
+		// Convert back to epoch time
+		roundedDownEpochTime := truncated.Unix()
+		startTime = uint32(roundedDownEpochTime)
+	}
+	for startTime <= endTime {
+		// only add timestamps which are multiple of 5 minutes
+		allTimestamps[startTime] = struct{}{}
+		startTime = startTime + calculatedInterval
+	}
```

Comment:
This code doesn't necessarily have anything to do with "5 minutes." Maybe
```
// Add timestamps every interval
```
would be better; or you could delete the comment since the code is pretty clear